src/views/ProfileView.vue

```diff
@@ -0,0 +1,86 @@
+<script>
+import { PROFILE_URL } from "../utility/UrlConstants.js";
+import {
+  appendAuthenticationHeaders
+} from "../utility/AuthenticationUtility.js";
+
+export default {
+  data() {
+    return {
+      userProfileData: null,
+      errorMsg: null,
+      isDataLoading: false,
+    }
+  },
+  methods: {
+    fetchData() {
+      this.userProfileData = null;
+      this.errorMsg = null;
+      this.isDataLoading = true;
+
+      const apiHeaders = appendAuthenticationHeaders({
+        "Content-Type": "application/json",
+      });
+
+      fetch(PROFILE_URL, {
+        method: "GET",
+        headers: apiHeaders,
+      })
+        .then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error("Unable to fetch user data");
+          }
+        })
+        .then((responseJson) => {
+          this.userProfileData = responseJson;
+          this.isDataLoading = false;
+        })
+        .catch((error) => {
+          this.isDataLoading = false;
+          this.errorMsg = error.message;
+          console.log(error);
+        });
+    },
+  },
+  mounted() {
+    this.fetchData();
+  },
+  created() {
+    // watch the params of this route to fetch this data again
+    this.$watch(
+      () => this.$route.params,
+      () => {
+        this.fetchData();
+      },
+      { immediate: true }
+    );
+  },
+};
+</script>
+<template>
+  <div class="container px-5 py-3" style="min-height: 60vh">
+    <h2 class="pb-3">Profile</h2>
+    <div v-if="errorMsg" class="alert alert-danger">{{ errorMsg }}
+    </div>
```

Comment:
```suggestion
    <div class="alert alert-danger mt-3" role="alert" v-if="errorMsg">
      <div><i class="bi bi-exclamation-circle-fill"></i> {{ errorMsg }}</div>
    </div>
```

Reply:
this is the error msg style we are following in other pages so just for consistency you can follow the same